### PR TITLE
Update MountsRarity into a single file

### LINKED ISSUE #5: possible Naming Conflict within Addon Namespace
Hi there :)

I'd like to use this Project as an embedded Libary in my Addon. By adding the MountsRarity-2.0.xml into my toc-File some Variables get mixed into my Addon Namespace. For me, that is still fine. BUT I think "data" and "MINOR_VERSION" are very generic names, which might collide with other Addons or even Libaries.
So at  the very least, please consider some more unique Variable Names. You could just use a unique Prefix, which might avoid this already completely.
A better way would be to avoid the Addon Namespace entirely. For that you would have to generate everything into a single file. I think that's technically feasible.

### PR DESCRIPTION
**Pull request by exochron**
This PR moves all data into the main script file. The CI updates that file using sed and some simple regex.
Furthermore the data table is loaded only when needed.

also fixes #5 :)

### CODE REVIEW

File: MountsRarity.lua [hunk 1]
```diff
@@ -20,20 +20,1109 @@
...
 if not MountsRarity then return end -- already loaded and no upgrade necessary
 
+local lazyLoadData = function()
+  return {}
+end
+local data
+
 function MountsRarity:GetData()
+  if not data then
+    data = lazyLoadData()
```
Comment (sgade) [inline MountsRarity.lua:36]: I very much like the idea of lazy loading the table.

File: scripts/update-rarities.sh [hunk 2]
```diff
@@ -28,15 +28,11 @@ MOUNTSRARITY_RESPONSE=$(get "$MOUNTSRARITY_SOURCE")
 MOUNTSRARITY=$(echo "$MOUNTSRARITY_RESPONSE" | jq '.mounts | to_entries')
 echo "Downloaded $(echo "$MOUNTSRARITY" | jq 'length') mounts."
 
+sed -i '/Everything after this line/q' MountsRarity.lua
```
Comment (sgade) [inline scripts/update-rarities.sh]: This should use the variable.

```suggestion
sed -i '/Everything after this line/q' $MOUNTS_FILE
```

**Review by sgade — CHANGES_REQUESTED** [review]
Sorry for the long time it took to review this PR. Thanks again for your contribution! I added one comment that you can please incorporate into this PR.

After that, you'd probably wanna rebase your changes on the current state because the `Data.lua` has been updated a few times in the meantime. Please make sure to copy over the current rarities into the new location. Thanks!

**Comment by exochron** [comment]
Done.
After merging you should probably manually run the update action as well, so that the data and version gets updated.

**Comment by sgade** [comment]
Awesome, merged! Thanks again for your contribution. Looking forward to trying out your addon.

**Pull request merged**